src/supervisor.sbvr

```diff
@@ -0,0 +1,461 @@
+Vocabulary: supervisor
+
+-- "(Type)" refers to the target database type to be autogenerated
+
+-- GENERIC TERMS
+
+Term: download progress
+	Concept Type: Integer (Type)
+
+Term: ip address
+	Concept Type: Short Text (Type)
+
+Term: name
+	Concept Type: Short Text (Type)
+
+Term: port
+	Concept Type: Integer (Type)
+
+Term: slug
+	Concept Type: Short Text (Type)
+
+Term: status
+	Concept Type: Short Text (Type)
+
+Term: version
+	Concept Type: Short Text (Type)
+
+Term: uuid
+	Concept Type: Text (Type)
+
+
+
+-- HOST CONFIG PROXY
+-- Discuss: we may not want to incorporate this into the Supervisor model
+-- These don't need to be a database types (the SV rewrites redsocks.conf under the hood) but not sure how to express that
+
+Term: proxy type
+	Concept Type: Short Text (Type)
+	Definition: "socks4" or "socks5" or "http-connect" or "http-relay"
+
+Term: login
+	Concept Type: Short Text (Type)
+
+Term: password
+	Concept Type: Hashed (Type)
+	-- SV doesn't implement hashing currently and it may not be necessary
+
+Term: no proxy
+	Synonym: ignore list
+	Fact type: no proxy has ip address
+		Necessity: each no proxy has zero or more ip addresses.
+
+Term: proxy
+
+	Fact type: proxy has proxy type
+		Necessity: each proxy has exactly one proxy type.
+	
+	Fact type: proxy has login
+		Necessity: each proxy has at most one login.
+
+	Fact type: proxy has password
+		Necessity: each proxy has at most one password.
+
+	Fact type: proxy has ip address
+		Necessity: each proxy has exactly one ip address.
+
+	Fact type: proxy has port
+		Necessity: each proxy has exactly one port.
+
+	Fact type: proxy has no proxy
+		Necessity: each proxy has at most one no proxy.
+
+
+
+-- DEVICE
+
+Term: mac address
+	Concept Type: Short Text (Type)
+
+Term: os variant
+	Concept Type: Short Text (Type)
+
+Term: hostname
+	Concept Type: Short Text (Type)
+	-- This doesn't need to be a database type (the SV calls the hostname service under the hood) but not sure how to express that
+
+Term: device
+
+	Fact type: device has ip address
+		Necessity: each device has at least one ip address.
+
+	Fact type: device has mac address
+		Necessity: each device has at least one mac address.
+
+	Fact type: device has status
+		Necessity: each device has exactly one status.
+		Definition: "Provisioning" or "Post-Provisioning" or "Stopping" or "Starting" or "Downloading" or "Installing" or "Idle" 
+		-- TODO: these are partially taken from v2 API reference which may not be ideal. Decide on status strings for v3.
+
+	Fact type: device has name
+		Necessity: each device has exactly one name.
+
+	Fact type: device has version
+		-- OS version
```

Comment:
I would maybe remove this, as we probably want to think about the future model where the OS is another app and the version belongs to the release of that app. Same for the os variant below